### DISCUSSION TITLE
cirrus-cli expected to work on windows?

### DISCUSSION THREAD

**Discussion by anarazel** [cirruslabs/cirrus-cli#558] [General]
Hi,

I tried to use cirrus-cli to locally debug a transient issue of a windows task on CI. Unfortunately that fails, due to connection issues:

```
| 'Windows - Server 2019, VS 2019 - Meson & msbuild' Task 06:19
  + Preparing execution environment... 02:03
  + image pull 01:25
  ...
  container: WARNING: 2022/09/07 22:22:22 [core] grpc: addrConn.createTransport failed to connect to {C:\Windows\Temp\c"  container: WARNING: 2022/09/07 22:22:22 [core] grpc: addrConn.createTransport failed to connect to {C:\Windows\Temp\c"  container: INFO: 2022/09/07 22:22:22 [core] Subchannel Connectivity change to TRANSIENT_FAILURE
  container: INFO: 2022/09/07 22:22:22 [core] Channel Connectivity change to TRANSIENT_FAILURE
```

Is that expected to work?

Regards,

Andres

**Comment by edigaryev** [comment]
Cirrus CLI works fine for me when using `windows_container:` on "Windows Server 2019 Datacenter for Containers" (GCP image, GCP VM) with the following YAML configuration:

```yaml
windows_container:
  image: cirrusci/windowsservercore:2019

task:
  script: true
```

It seems weird that the RPC server path got truncated to `C:\Windows\Temp\c` in your logs, though.

Could you at least provide some minimally failing configuration and a description of the environment you're trying to run the Cirrus CLI in?

**Reply by anarazel** [reply]
> Cirrus CLI works fine for me when using windows_container: on "Windows Server 2019 Datacenter for Containers" (GCP image, GCP VM) with the following YAML configuration

Good to know. I mostly didn't know whether to prepare a more detailed bug report or whether it's just unsupported, based on the language in the README.md:

> Note: Cirrus CLI only supports [Linux container](https://cirrus-ci.org/guide/linux/#linux-containers) and [macos_instance VMs](https://cirrus-ci.org/guide/macOS/) at the moment. Linux containers support the [Dockerfile as a CI environment](https://cirrus-ci.org/guide/docker-builder-vm/#dockerfile-as-a-ci-environment) feature.


> It seems weird that the RPC server path got truncated to C:\Windows\Temp\c in your logs, though.

I think that's just cirrus-cli truncating log messages due to the terminal's width being reached. If I look at docker logs of the started container the messages aren't truncated:

```
2022/09/12 18:35:32 Running agent version 1.73.3-d6db039
INFO: 2022/09/12 18:35:32 [core] parsed scheme: "unix"
INFO: 2022/09/12 18:35:32 [core] ccResolverWrapper: sending update to cc: {[{C:\Windows\Temp\cli-18ef81cc-0b70-43a2-b9c4-87a1eb20c13d\cli.sock  0xc00008c820 0 <nil>}] <nil> <nil>}
INFO: 2022/09/12 18:35:32 [core] ClientConn switching balancer to "pick_first"
INFO: 2022/09/12 18:35:32 [core] Channel switches to new LB policy "pick_first"
INFO: 2022/09/12 18:35:32 [core] Subchannel Connectivity change to CONNECTING
INFO: 2022/09/12 18:35:32 [core] Subchannel picks a new address "C:\\Windows\\Temp\\cli-18ef81cc-0b70-43a2-b9c4-87a1eb20c13d\\cli.sock" to connect
INFO: 2022/09/12 18:35:32 [core] Channel Connectivity change to CONNECTING
WARNING: 2022/09/12 18:35:32 [core] grpc: addrConn.createTransport failed to connect to {C:\Windows\Temp\cli-18ef81cc-0b70-43a2-b9c4-87a1eb20c13d\cli.sock localhost 0xc00008c820 0 <nil>}. Err: connection error: desc = "transport: Error while dialing dial unix C:\\Windows\\Temp\\cli-18ef81cc-0b70-43a2-b9c4-87a1eb20c13d\\cli.sock: connect: No connection could be made because the target machine actively refused it."
WARNING: 2022/09/12 18:35:32 [core] grpc: addrConn.createTransport failed to connect to {C:\Windows\Temp\cli-18ef81cc-0b70-43a2-b9c4-87a1eb20c13d\cli.sock localhost 0xc00008c820 0 <nil>}. Err: connection error: desc = "transport: Error while dialing dial unix C:\\Windows\\Temp\\cli-18ef81cc-0b70-43a2-b9c4-87a1eb20c13d\\cli.sock: connect: No connection could be made because the target machine actively refused it."
INFO: 2022/09/12 18:35:32 [core] Subchannel Connectivity change to TRANSIENT_FAILURE
INFO: 2022/09/12 18:35:32 [core] Channel Connectivity change to TRANSIENT_FAILURE
INFO: 2022/09/12 18:35:33 [core] Subchannel Connectivity change to IDLE
INFO: 2022/09/12 18:35:33 [core] Channel Connectivity change to IDLE
INFO: 2022/09/12 18:35:33 [core] Subchannel Connectivity change to CONNECTING
```

entering the container shows 'C:\\Windows\\Temp\\cli-18ef81cc-0b70-43a2-b9c4-87a1eb20c13d\\cli.sock: connect' to exist - not sure how to check it can actually be connected to.

> Could you at least provide some minimally failing configuration

This is inside a windows 10 VM (with nested virtualization support) with docker installed and running. As far as I can tell the details of the task don't matter - your example from above fails too.

**Reply by edigaryev** [reply]
Are you sure you are running Windows containers and not Docker Desktop?

**Reply by anarazel** [reply]
Fairly - docker desktop is not installed, I started dockerd manually on the commandline, and I can create and use windows containers.
